common/decisions/00_major_decisions_german_culture.txt

```diff
@@ -0,0 +1,70 @@
+form_german_culture_decision = {
+	picture = {
+		reference = "gfx/interface/illustrations/decisions/decision_dynasty_house.dds"
+	}
+	desc = form_german_culture_decision_desc
+	selection_tooltip = form_german_culture_decision_tooltip
+	decision_group_type = major
+
+	ai_check_interval = 120
+
+	is_shown = {
+		is_ruler = yes
+		is_playable_character = yes
+		highest_held_title_tier > 3
+
+		root = {
+			OR = {
+				culture = culture:franconian
+				culture = culture:saxon
+				culture = culture:swabian
+				culture = culture:bavarian
+				culture = culture:czech
+				culture = culture:polabian
+				culture = culture:pomeranian
+				culture = culture:polish
+				culture = culture:carantanian
+				culture = culture:hungarian
+				culture = culture:croatian
+			}
+		}
+	}
+
+	is_valid = {
+		AND = {
+			completely_controls_region = german_unified_lands
+		}
+	}
+
+	is_valid_showing_failures_only = {
+		is_landed = yes
+		is_capable_adult = yes
+		is_imprisoned = no
+		is_at_war = no
+	}
+
+	effect = {
+		gain_heroic_legend_seed_tooltip_effect = yes
+		custom_tooltip = form_german_culture_decision_effect_tooltip
+		
+		hidden_effect = {
+
+		}
+
+		convert_family_culture_and_notify_vassals_effect = {
+			CONVERTER = root
+			OLD_CULTURE = root.culture
+			NEW_CULTURE = culture:unified_german
+		}
+
+		discover_all_eras = all
```

Comment:
`discover_all_eras` looks incorrect here, as I would assume that unlocks everything.

Here's some code I found which allowed updating the culture innovations:
```
every_culture_global = {
    limit = { has_cultural_pillar = heritage_latin }

    add_innovation_if_missing_effect = { INNOVATION = innovation_men_at_arms }
    add_innovation_if_missing_effect = { INNOVATION = innovation_pike_columns }
    add_innovation_if_missing_effect = { INNOVATION = innovation_guilds }
}
```

Obviously, instead of `limit = { has_cultural_pillar = heritage_latin }` you'll want `limit = { culture = unified_german }`

Hence in theory you'd need something like:
```
every_culture_global = {
   # We only care about the culture the root character currently is
   limit = {
      culture = root.culture
   }
   save_scope_as = original_culture

   # Update all innovations for German to at least have those of the enactor's culture
   every_culture_group = {
      limit = {
         culture = unified_german
      }

     if = {
           limit = {
                scope:original_culture = {
                        has_innovation = innovation_quilted_armor
                }
            }
           add_innovation_if_missing_effect = { INNOVATION = innovation_quilted_armor }
    }
   # ... repeated for every innovation, to ensure it works test with a handful first
   }
}
```

Reply:
Regarding this, I saw a code that will add innovations from culture to culture which is: 

			root.culture = {
				get_all_innovations_from = culture:unified_german
			}

Reply:
Great find!
I looked up that in the base game codebase and saw it's usage is like so:

<img width="358" alt="Screenshot 2024-11-14 at 15 50 28" src="https://github.com/user-attachments/assets/3008b388-0c19-4656-8250-aba85aa2918e">

Hence I've added another comment with a suggested re-write for this bit of code